## 1. Introducción

##### hola



## 2. Importación de librerias


In [1]:
import pandas as pd
import numpy as np

from pathlib import Path

import matplotlib.pyplot as plt



## 3. Carga de datos

In [ ]:
RUTA = Path("./Data")

dim_conductor = pd.read_csv(RUTA / "dim_conductor.csv",
                             sep=",")
dim_depot = pd.read_csv(RUTA / "dim_depot.csv")
dim_linea = pd.read_csv(RUTA / "dim_linea.csv")
dim_parada = pd.read_csv(RUTA / "dim_parada.csv")
dim_tarifa = pd.read_csv(RUTA / "dim_tarifa.csv")
dim_vehiculo = pd.read_csv(RUTA / "dim_vehiculo.csv")
 
fact_incidencias = pd.read_csv(RUTA / "fact_incidencias.csv")
fact_mantenimiento = pd.read_csv(RUTA / "fact_mantenimiento.csv")
fact_viajes = pd.read_csv(RUTA / "fact_viajes.csv")

## 4. Exploración inicial


### - dimensiones

##### En este apartado se realiza una primera revisión de cada una de las tablas del conjunto de datos con el objetivo de conocer su estructura, el volumen de información disponible y los tipos de datos de cada variable. Esta exploración permitirá detectar posibles incidencias antes de comenzar el análisis exploratorio.


In [3]:
tablas = {
    "dim_conductor": dim_conductor,
    "dim_depot": dim_depot,
    "dim_linea": dim_linea,
    "dim_parada": dim_parada,
    "dim_tarifa": dim_tarifa,
    "dim_vehiculo": dim_vehiculo,
    "fact_incidencias": fact_incidencias,
    "fact_mantenimiento": fact_mantenimiento,
    "fact_viajes": fact_viajes
}

for nombre, tabla in tablas.items():
    print(f"{nombre}: {tabla.shape}")

dim_conductor: (30, 1)
dim_depot: (3, 6)
dim_linea: (10, 1)
dim_parada: (120, 10)
dim_tarifa: (9, 1)
dim_vehiculo: (45, 1)
fact_incidencias: (4000, 1)
fact_mantenimiento: (876, 15)
fact_viajes: (50000, 24)


##### Interpretación
##### Con estos resultados vemos que las tablas de dimensiones contienen un número reducido registros, a excepción de dim_paradas que siendo reducido aporta 120 registros muy superior a las demas tablas de dimensiones. Mientras que las tablas de de hechos contienen un número muy elevado llegando hasta los 50000 registros en una tabla, lo cual es coherente con la estructura de un modelo dimensional destinado al análisis de datos.

### - tipos
#### En este apartado vamos a valorar los tipos de datos que nos aportan las tablas.

In [11]:
for nombre, tabla in tablas.items():
    print("="*60)
    print(nombre)
    tabla.info()

dim_conductor
<class 'pandas.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 1 columns):
 #   Column                                                                                                                        Non-Null Count  Dtype
---  ------                                                                                                                        --------------  -----
 0   conductor_id,nombre,anno_incorporacion,antiguedad_anos,turno_habitual,depot_id,formacion,licencia_tipo,activo,ausencias_2024  30 non-null     str  
dtypes: str(1)
memory usage: 2.3 KB
dim_depot
<class 'pandas.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   depot_id             3 non-null      int64  
 1   nombre               3 non-null      str    
 2   barrio               3 non-null      str    
 3   latitud              3 non-null      float64
 

#### Interpretación
#### Viendo los resultados podemos apreciar que hay varias columnas en diferentes tablsa que deberian de ser bool y sin embargo son object o str, lo curioso es que también hay columnas de bool. Lo que me llama mucho la tención que todas las consideras str u object son verdadero/falto a execepción de una que es true/false, el resto que aparecen como true/false son bool. Por último, hay una columna fecha que aparece como str y no como datetime, supongo que sera igual y el motivo es por como está escrito.

### - primeras filas
#### En este apartado vamos a analizar las primeras filas y ver que los valores y las columnas sean acordes a la tabla.

In [34]:
for nombre, tabla in tablas.items():
    print("="*60)
    print(nombre)
    display(tabla.head())

dim_conductor


,conductor_id,nombre,anno_incorporacion,antiguedad_anos,turno_habitual,depot_id,formacion,licencia_tipo,activo,ausencias_2024
0,1,Carlos Garcia,2009,15,Partido,3,Basica,D,VERDADERO,14
1,2,Maria Lopez,2013,11,Noche (22-06h),1,Basica,D,VERDADERO,9
2,3,Juan Martinez,2008,16,Manana (06-14h),2,Completa,D+E,VERDADERO,6
3,4,Ana Fernandez,2007,17,manana,1,Basica + Articulado,D,VERDADERO,1
4,5,Pedro Sanchez,2016,8,Partido,3,Basica + Articulado,D,VERDADERO,10


dim_depot


,depot_id,nombre,barrio,latitud,longitud,capacidad_vehiculos
0,1,Cochera Norte,Barrio Norte,40.440,-3.720,20
1,2,Cochera Central,Centro,40.418,-3.700,15
2,3,Cochera Sur,Barrio Sur,40.395,-3.695,12


dim_linea


,linea_id,codigo,nombre,tipo,km_recorrido,n_paradas,frecuencia_min
0,1,L1,Centro - Aeropuerto,Urbana,"18,4",32,12
1,2,L2,Universidad - Hospital,Urbana,"11,2",21,8
2,3,L3,Barrio Norte - Estacion,Urbana,"9,7",18,10
3,4,L4,Poligono Industrial - Centro,Urbana,"14,1",24,15
4,5,L5,Circular Centro,Urbana,"7,3",14,6


dim_parada


,parada_id,nombre_parada,barrio,tipo,latitud,longitud,accesible_silla,marquesina,panel_informacion,activa
0,1,Parada Barrio Norte 1,Barrio Norte,Intermedia,40.431989,-3.699183,True,False,False,True
1,2,Parada Barrio Sur 2,Barrio Sur,Intermedia,40.403568,-3.692632,True,True,True,True
2,3,Parada Universidad 3,Universidad,Intermedia,40.419744,-3.723256,True,False,False,True
3,4,Parada Hospital 4,Hospital,Intermedia,40.410394,-3.689212,False,False,True,True
4,5,Parada Poligono 5,centro,Intermedia,40.410427,-3.715006,False,False,True,True


dim_tarifa


,tarifa_id,tipo_titulo,categoria,precio_eur,es_abono,bonificado
0,1,Ordinario,Adulto,15,FALSO,FALSO
1,2,Bono 10 Viajes,Adulto,97,VERDADERO,FALSO
2,3,Abono Mensual,Adulto,6,VERDADERO,FALSO
3,4,Abono Joven,Joven,4,VERDADERO,VERDADERO
4,5,Abono Jubilado,Jubilado,25,VERDADERO,VERDADERO


dim_vehiculo


,vehiculo_id,matricula,modelo,combustible,capacidad_sentados,capacidad_total,anno_fabricacion,anno_incorporacion,km_totales,depot_id,emisiones_co2_gkm,en_servicio
0,1,6001 BUS,Mercedes-Benz Citaro,Diesel,88,128,2017,2018,586265,1,127,VERDADERO
1,2,6002 BUS,Mercedes-Benz Citaro,Diesel,88,128,2018,2019,317095,1,112,VERDADERO
2,3,6003 BUS,Iveco Urbanway,diesel,90,130,2018,2019,256606,2,133,VERDADERO
3,4,6004 BUS,Iveco Urbanway,Diesel,90,130,2019,2020,379164,3,121,VERDADERO
4,5,6005 BUS,Solaris Urbino 12,Diesel,85,125,2014,2015,581756,2,119,VERDADERO


fact_incidencias


,incidencia_id,viaje_id,vehiculo_id,conductor_id,linea_id,fecha,anno,mes,hora_incidencia,tipo_incidencia,categoria,severidad,requiere_retirada,duracion_resolucion_dias,vehiculo_sustituto,coste_estimado_eur
0,1,33554,39,11,4,01/06/2022,2022,6,19:11:00,Huelga parcial,Operacional,Alta,VERDADERO,60,FALSO,0
1,2,9428,21,21,6,16/02/2023,2023,2,20:42:00,Fallo electrico,Vehiculo,Alta,VERDADERO,62,FALSO,57551
2,3,200,33,5,8,20/01/2022,2022,1,6:21:00,Accidente leve,Seguridad,Alta,VERDADERO,29,VERDADERO,33180
3,4,12448,17,17,10,16/01/2023,2023,1,2:30:00,Vandalismo,Seguridad,Media,FALSO,16,FALSO,65199
4,5,39490,11,11,6,08/01/2023,2023,1,14:51:00,Retencion trafico,Externo,Baja,FALSO,14,FALSO,4988


fact_mantenimiento


,mantenimiento_id,vehiculo_id,depot_id,fecha_entrada,fecha_salida,anno,mes,tipo_mantenimiento,categoria,es_correctivo,dias_fuera_servicio,km_en_revision,coste_eur,proveedor,garantia_meses
0,1,1,1,2024-07-24,2024-07-25,2024,7,Cambio aceite,Preventivo,False,1,278454,218.70,TallerBus Norte,0
1,2,1,1,2023-02-08,2023-02-09,2023,2,Cambio neumaticos,Preventivo,False,1,240059,1068.93,TallerBus Norte,0
2,3,1,1,2024-10-28,2024-10-29,2024,10,Cambio neumaticos,Preventivo,False,1,493071,722.19,Taller Oficial Mercedes,0
3,4,1,1,2022-08-01,2022-08-03,2022,8,Reparacion clima,Correctivo,True,2,533556,579.93,FlotaService,6
4,5,1,1,2024-12-09,2024-12-10,2024,12,Revision bateria,Preventivo,False,1,307110,436.54,TallerBus Norte,0


fact_viajes


,viaje_id,linea_id,vehiculo_id,conductor_id,parada_origen_id,parada_destino_id,fecha,anno,mes,dia_semana,...,hora_llegada_real,retraso_salida_min,duracion_real_min,pasajeros_subidos,ocupacion_pct,km_programados,km_recorridos,viaje_completado,consumo,tarifa_predominante_id
0,1,4,2,2,17,116,2023-08-14,2023,8,Monday,...,19:22,19,73,78.0,0.609,14.1,14.1,True,4.69,4
1,2,4,3,3,5,48,2024-05-20,2024,5,Monday,...,08:30,13,67,115.0,0.885,14.1,14.1,True,5.86,2
2,3,5,4,4,33,110,2023-08-15,2023,8,Tuesday,...,13:05,0,35,47.0,0.362,7.3,7.3,True,3.15,3
3,4,2,5,5,37,116,2022-03-23,2022,3,Wednesday,...,21:36,0,46,111.0,0.888,11.2,11.2,True,3.50,9
4,5,5,6,6,12,115,2022-03-24,2022,3,Thursday,...,18:50,8,32,75.0,0.405,7.3,7.3,True,3.13,3


#### Interpretación
#### En este apartado podemos apreciar como los valores con acordes a las columnas. Las columnas estan bien colocadas y no apelotonas ya que hemos usado en algunas de ellas sep=";".

### Conclusión de la exploración incial
#### Tras revisar las dimensiones, los tipos de datos y las primeras filas de todas las tablas se observa que el conjunto de datos presenta una estructura coherente y adecuada para su análisis.

#### No se han detectado valores nulos ni duplicados en las claves primarias de las tablas de dimensión. Sin embargo, se han identificado algunas columnas cuyos tipos de datos podrían optimizarse durante la fase de limpieza, especialmente variables booleanas almacenadas como texto y columnas de fechas importadas como cadenas de caracteres.

## 5. Calidad de datos

### - valores nulos

### - duplicados

### - tipos incorrectos

## 6. Análisis estadistico


## 7. Outliers


## 8. Visualizaciones

## 9. Conclusiones